packages/client/src/router/resolveRoutePath.ts

```diff
@@ -1,12 +1,22 @@
 import { normalizeRoutePath } from '@vuepress/shared'
 import { redirects, routes } from '../internal/routes.js'
 
+const FAKE_HOST = 'http://.'
+
 /**
  * Resolve route path with given raw path
  */
-export const resolveRoutePath = (path: string): string => {
+export const resolveRoutePath = (path: string, current?: string): string => {
+  let routePath = path
+
+  if (!path.startsWith('/') && current) {
+    const loc = current.slice(0, current.lastIndexOf('/'))
+
+    routePath = new URL(`${loc}/${path}`, FAKE_HOST).pathname
+  }
```

Comment:
Better to extract to a standalone file like `resolveRelativePath`. Maybe put it to shared pkg and add unit tests for it.